src/payment/strategies/masterpass/masterpass-payment-strategy.ts

```diff
@@ -0,0 +1,121 @@
+import { PaymentStrategy } from '../';
+import {
+    PaymentActionCreator,
+    PaymentInitializeOptions,
+    PaymentMethodActionCreator,
+    PaymentRequestOptions
+} from '../../';
+import { Checkout, CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import {
+    InvalidArgumentError,
+    MissingDataError,
+    MissingDataErrorType
+} from '../../../common/error/errors';
+import { StoreConfig } from '../../../config';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import PaymentMethod from '../../payment-method';
+
+import { MasterpassCheckoutCallback, MasterpassCheckoutOptions } from './masterpass';
+import MasterpassScriptLoader from './masterpass-script-loader';
+
+export default class MasterpassPaymentStrategy extends PaymentStrategy {
+    private _paymentMethod?: PaymentMethod;
+
+    constructor(
+        store: CheckoutStore,
+        private _orderActionCreator: OrderActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _masterpassScriptLoader: MasterpassScriptLoader
+    ) {
+        super(store);
+    }
+
+    initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const { methodId } = options;
+
+        this._paymentMethod = this._store.getState().paymentMethods.getPaymentMethod(methodId);
+
+        if (!this._paymentMethod) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        if (!this._hasPaymentInfo()) {
+            return this._masterpassClientSetup().then(checkoutCallback => checkoutCallback()).then(() => super.initialize(options));
```

Comment:
Could you please move every `then` to a different line?

```js
return this._masterpassClientSetup()
	.then(checkoutCallback => checkoutCallback())
	.then(() => super.initialize(options));
```

Also, since all you are doing with the callback is calling it, it might be a good idea to make it part of `_masterpassClientSetup`... what do you think?